gobblin-compaction/src/main/java/gobblin/compaction/hivebasedconstructs/CompactionLauncherWriter.java

```diff
@@ -0,0 +1,53 @@
+package gobblin.compaction.hivebasedconstructs;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import gobblin.compaction.listeners.CompactorListener;
+import gobblin.compaction.mapreduce.MRCompactor;
+import gobblin.metrics.Tag;
+import gobblin.writer.DataWriter;
+import gobblin.compaction.mapreduce.avro.ConfBasedDeltaFieldProvider;
+
+
+/**
+ * {@link DataWriter} that launches an {@link MRCompactor} job given an {@link MRCompactionEntity} specifying config
+ * for compaction
+ */
+public class CompactionLauncherWriter implements DataWriter<MRCompactionEntity> {
+  @Override
+  public void write(MRCompactionEntity compactionEntity) throws IOException {
+    Preconditions.checkNotNull(compactionEntity);
+
+    List<? extends Tag<?>> list = new ArrayList<>();
+
+    Properties props = compactionEntity.getProps();
+    props.setProperty(ConfBasedDeltaFieldProvider.DELTA_FIELDS_KEY, compactionEntity.getDelta());
```

Comment:
Since you are using `ConfBasedDeltaFieldProvider`, please add it to the java doc.